internal/pkg/agent/install/uninstall.go

```diff
@@ -99,27 +99,23 @@ func Uninstall(cfgFile, topPath, uninstallToken string) error {
 func RemovePath(path string) error {
 	const arbitraryTimeout = 5 * time.Second
 	start := time.Now()
-	nextSleep := 1 * time.Millisecond
-	for {
-		err := os.RemoveAll(path)
-		if err == nil {
-			return nil
-		}
-		if isBlockingOnExe(err) {
-			// try to remove the blocking exe
-			err = removeBlockingExe(err)
-		}
-		if err == nil {
-			return nil
-		}
-		if !isRetryableError(err) {
-			return err
+	var lastErr error
+	for time.Since(start) <= arbitraryTimeout {
+		lastErr = os.RemoveAll(path)
+
+		if lastErr == nil || !isRetryableError(lastErr) {
+			return lastErr
 		}
 
-		if d := time.Since(start) + nextSleep; d >= arbitraryTimeout {
-			return err
+		if isBlockingOnExe(lastErr) {
+			// try to remove the blocking exe and try again to clean up the path
+			removeBlockingExe(lastErr)
 		}
+
+		time.Sleep(time.Millisecond)
```

Comment:
IMHO we should sleep longer. Ignoring the time it takes to run the commands above, this line will be called 5000 times, which is a whole lot. How about 50msec?

Reply:
well before the change we were not even sleeping, we were spinning in the loop. Than can be easily inclreased... (also on windows I think the minimum clock tick is ~15 ms)

Reply:
Done